example/autoencoder/convolutional_autoencoder.ipynb

```diff
@@ -0,0 +1,578 @@
+      "text/plain": [
+       "<Figure size 1440x288 with 20 Axes>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "plt.figure(figsize=(20,4))\n",
+    "for i in range(10):\n",
+    "    idx = random.randint(0, len(test_dataset))\n",
+    "    img, _ = test_dataset[idx]\n",
+    "    x, _ = test_dataset_t[idx]\n",
+    "\n",
+    "    data = x.as_in_context(ctx).expand_dims(axis=0)\n",
+    "    output = net(data)\n",
+    "
```

Comment:
Wouldn't it be nicer to create these as separate HybridSequential blocks in the first place? And then AutoEncoder would be a HybridSequential block of encoder and decoder, and you can just reference encoder and decoder here. Also covers case when user changes the encoder and screws up the layer numbering.

Reply:
agreed :+1: